### PR TITLE
refactor: Move m_num_unconnecting_headers_msgs out of cs_main guard

### PR DESCRIPTION
**Pull request by maflcko**
The variable does not need to be guarded by cs_main, so move it out of the cs_main scope. It is not used for validation-specific state, but only for headers-processing internal state and logging.

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -2221,28 +2222,26 @@ void PeerManagerImpl::SendBlockTransactions(CNode& pfrom, Peer& peer, const CBlo
  *
  * We'll send a getheaders message in response to try to connect the chain.
  *
- * The peer can send up to MAX_UNCONNECTING_HEADERS in a row that
+ * The peer can send up to MAX_UNCONNECTING_HEADERS_MSGS in a row that
```
Comment (sdaftuar) [inline src/net_processing.cpp:2225]: Care to address [this comment ](https://github.com/bitcoin/bitcoin/pull/25454/files/99f4785cad94657dcf349d00fdd6f1d44cac9bb0#r912929185) while you're touching this line?

File: src/net_processing.cpp [hunk 2]
```diff
@@ -250,6 +250,9 @@ struct Peer {
       * to the peer. */
     std::chrono::microseconds m_next_send_feefilter{0};
 
+    //! Length of current-streak of unconnecting headers announcements
+    int num_unconnecting_headers_msgs{0};
```
Comment (sdaftuar) [inline src/net_processing.cpp:254]: Should this be named `m_num_unconnecting_headers_msgs` for consistency with the style guide?

Comment (MarcoFalke) [reply src/net_processing.cpp:254]: `Peer` is a `struct`, so it shouldn't use `m_` prefixes. To quote the style guide:

> Class member variables have a m_ prefix.

For rationale, see for example https://github.com/bitcoin/bitcoin/pull/19759

(So in theory https://github.com/bitcoin/bitcoin/pull/25557 could drop the `m_` prefix)

Though, I don't care about the style guide too much, so I am happy to change this to make reviewers happy and I am also not going to comment on https://github.com/bitcoin/bitcoin/pull/25557 to suggest a change.

File: src/net_processing.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/net_processing.cpp:2225]: Thanks, done.

**Comment by achow101** [comment]
Are you still working on this?

**Comment by MarcoFalke** [comment]
Rebased, but closing for now to pick it up later

**Pull request closed without merging**